wand-watcher.lic

```diff
@@ -0,0 +1,153 @@
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#wand-watcher
+=end
+
+no_kill_all
+no_pause_all
+
+custom_require.call(%w[common common-items])
+
+class WandWatcher
+  include DRC
+  include DRCI
+
+  def initialize
+    settings = get_settings();
+    @wand_list = settings.wands
+    @queued_wands = []
+    @no_use_scripts = settings.wand_watcher_no_use_scripts || []
+    @startup_delay = settings.wand_watcher_startup_delay || 10
+    @passive_delay = settings.wand_watcher_passive_delay || 60
+    @activation_failure_messages = [/^The .* remains inert/]
+    UserVars.wand_watcher_timers = {} unless UserVars.wand_watcher_timers
+
+    #don't allow users to make foolish mistakes, always prevent use during go2/burgle
+    @no_use_scripts << 'go2'
+    @no_use_scripts << 'burgle'
+
+    # rebuild UserVars based on what wands are listed in the yaml
+    # this preserves timers existing wands, removes timers for wands no longer in the yaml
+    # and sets new wand next use timers to now
+    temp_timers = {}
+    @wand_list.each do |key,item|
+      if UserVars.wand_watcher_timers[key]
+        temp_timers[key] = UserVars.wand_watcher_timers[key]
+      else
+        temp_timers[key] = Time.now
+      end
+    end
+    UserVars.wand_watcher_timers = temp_timers
+    
+    #startup delay so you can control spammyness of autostart scripts
+    pause @startup_delay
+
+    #run passively
+    passive_run
+    #TODO: run once option
+  end
+
+  def passive_run
+    loop do
+      if @wand_list == nil || @wand_list.empty?
+        #@wand_list = nil when no wands configured in yaml
+        #@wand_list = empty when all wands are deleted due to problems finding wands
+        message "No wands to check.  Ending script.  Double check your settings."
+        exit
+      end
+      check_timers
+      pause @passive_delay
+    end
+  end
+
+  def check_timers  
+    #if any no_use_scripts are running, loop every 10 seconds checking until not
+    loop do
+      break if !(@no_use_scripts.any? { |name| Script.running?(name) })
+      pause 10
+    end
+    
```

Comment:
Proposing list of rooms where not to try and use the wands, such as banks and vaults or other areas where magical devices are restricted. This loops and checks every 10 seconds if the user has moved to a different room before resuming the script just like waiting for no_use_scripts to finish.
```suggestion

    #if player is in any no_use_rooms, loop every 10 seconds checking until not
    loop do
      # if in a room without an id then Room.current is nil but DRRoom.title is reliable
      break if !(@no_use_rooms.any? { |name| /#{name}/ =~ DRRoom.title || name.to_s == Room.current.id.to_s })
      pause 10
    end

```